doc/overview/breaking_changes_visual_studio_2019.md

```diff
@@ -0,0 +1,119 @@
+
+The following interfaces no longer inherit from `IProjectVersionedValue`:
+
+| Interface |
+| --- |
+| IProjectSnapshot |
+| IProjectRuleSnapshot |
+| IProjectSharedFoldersSnapshot |
+| IProjectImportTreeSnapshot |
+
+In most cases, we added a new `Value` property to the interface; originally it was inherited from `IProjectVersionedValue<T>`.
+
+Any code using the version number in those snapshots should use the version number in the dataflow directly.
+
+`IProjectSnapshot.Value` has been marked as obsolete, please use `IProjectSnapshot.ProjectInstance`.
+
+This change was made to have a consistent style in all dataflow services and to reuse unchanged immutable objects to reduce allocations (instead of creating new instances that include version only changes).
+
+## IBuildLoggerProvider has been marked as obsolete, and is no longer called
+
+Code should be changed to use `IBuildLoggerProviderAsync` instead
+
+## IVsLoggerEventProcessor is removed
+
+It was marked as obsolete in the previous release, and not called by the product.
+
+Mitigation: use `IVsLoggerEventProcessor2` instead
+
+## Property ActiveConfigurationChangedEventArgs.ActiveConfiguredProjectProviderDataSourceVersion is removed
+
+It was marked as obsolete in previews version.
+
+## Enum value ProjectFaultSeverity.Crippling has been removed.
+
+It was renamed to `LimitedFunctionality` in Visual Studio 2017 updates, and the old enum value has been marked as obsolete for a while.
+
+## LazyFill value has been removed from IProjectTreeSnapshot and TreeUpdateResult
+
+This value was not used and maintained since Dev 14.
+
+## IProjectTree.PropertySheet is marked as obsolete and no longer saved in the tree
+
+We don't expect it to be used.
+
+## IProjectLockService introduces a new set of delegate based API to replace the current using style API to avoid potential deadlocks
+
+The old using style API has not been removed or marked as obsolete at this point of time, so no action is required.
+
+But we will eventually mark them as obsolete later, so plan some time to move to the new API after Preview 1.
+
+Before:
+```csharp
+using (var access = await this.ProjectLockService.ReadLockAsync(cancellationToken))
+{
+    var project = await access.GetProjectAsync(this.ConfiguredProject);
+    // Use project...
+}
+```
+	
+After:
+```csharp
+await this.ProjectLockService.ReadLockAsync(
+    async access =>
+    {
+        Project project = await access.GetProjectAsync(this.ConfiguredProject);
+        // Use project...
+    },
+    cancellationToken);
+```
+
+## New Slim DataFlow blocks
+
+To reduce the memory usage of dataflow blocks, CPS introduces a new custom set of slim dataflow blocks corresponding to `TransformBlock`/`ActionBlock`/`BroadcastBlocks`.
+While this is not a breaking change, we recommend CPS extensions to start using the new blocks.
+
+The new blocks allow CPS dataflow to skip middle versions easier, but may introduce some behavior changes.
```

Comment:
In what cases would a version be skipped? My mental model is that all updates are queued. Is there some conflation going on? Is this only for the slim versions, or did the prior blocks do this too?